cmd/kubeadm/app/cmd/init_test.go

```diff
@@ -349,3 +363,26 @@ func expectedInitIgnorePreflightErrors(expectedItems ...string) func(t *testing.
 		}
 	}
 }
+
+func TestInitDataClientWithNonDefaultKubeconfig(t *testing.T) {
+	kubeconfigPath := filepath.Join(t.TempDir(), "custom.conf")
+	if err := os.WriteFile(kubeconfigPath, []byte(testKubeconfigData), 0o600); err != nil {
+		t.Fatalf("os.WriteFile returned unexpected error: %v", err)
+	}
+
+	// initialize an external init option and inject it to the init cmd
+	initOptions := newInitOptions()
+	initOptions.skipCRIDetect = true // avoid CRI detection in unit tests
+	initOptions.kubeconfigPath = kubeconfigPath
+	cmd := newCmdInit(nil, initOptions)
+
+	data, err := newInitData(cmd, nil, initOptions, nil)
+	if err != nil {
+		t.Fatalf("newInitData returned unexpected error: %v", err)
+	}
+
+	// This only works if the client that is constructed from the kubeconfig is not used.
+	if _, err := data.Client(); err != nil {
+		t.Fatalf("data.Client returned unexpected error: %v", err)
+	}
```

Comment:
can we actually verify that the initOptions.kubeconfigPath is used here?
one way would be with httptest, i guess.

Reply:
something like this?

```go
	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if r.Method != http.MethodHead {
			w.WriteHeader(http.StatusMethodNotAllowed)
		}
	}))
	defer ts.Close()

...

	result := client.Discovery().RESTClient().Verb("HEAD").Do(context.Background())
	if err := result.Error(); err != nil {
		t.Fatalf("REST client request returned unexpected error: %v", err)
	}
```

Reply:
SGTM!